File: MVVMJavaRx/app/src/unitTests/java/com/vascome/fogtail/FogtailRobolectricUnitTestRunner.java
package com.vascome.fogtail;

import android.support.annotation.NonNull;

import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

// Custom runner allows us set config in one place instead of setting it in each test class.
public class FogtailRobolectricUnitTestRunner extends RobolectricTestRunner {

    // This value should be changed as soon as Robolectric will support newer api.
    private static final int SDK_EMULATE_LEVEL = 23;

    public FogtailRobolectricUnitTestRunner(@NonNull Class<?> classRun) throws Exception {
        super(classRun);
    }

    @NonNull
    public static FogtailApplication fogtailApplication() {
        return (FogtailApplication) RuntimeEnvironment.application;
    }
}

File: MVVMKotlinArchComponent/app/src/unitTests/java/com/vascome/fogtail/FogtailRobolectricUnitTestRunner.kt
package com.vascome.fogtail

import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment

// Custom runner allows us set config in one place instead of setting it in each test class.
class FogtailRobolectricUnitTestRunner @Throws(Exception::class)
constructor(classRun: Class<*>) : RobolectricTestRunner(classRun) {
    companion object {

        // This value should be changed as soon as Robolectric will support newer api.
        private val SDK_EMULATE_LEVEL = 23

        fun fogtailApplication(): FogtailApplication {
            return RuntimeEnvironment.application as FogtailApplication
        }
    }
}
